### DISCUSSION TITLE
deployment on a personnal shiny server

### DISCUSSION THREAD

**Discussion by hedjour** [ThinkR-open/golem#682] [Q&A]
Hi
Is there in Golem a function to deploy simply our application on a personnal server ?
many thanks for all.

**Comment by VincentGuyader** [comment]
Hi,

please have a look to `golem::add_shinyserver_file()`

this will create an app.R to use to deploy on Shiny server.

regards

**Comment by hedjour** [comment]
Many thanks, I will look today
Regards

**Comment by ColinFay** [comment]
Hey, 

There are two strategies to deploy on shiny server: 

- Build the package, install it to the remote server, then write in your app.R: 

```r
library(myapp)
run_app()
```

- use `golem::add_shinyserver_file()`, then copy the whole folder to the remote server, where you want the app to be deployed 

To make this a little bit more smooth, I suggest using git to do the second option. 
That way, you can simply go on your server and do a `git pull` of the updated version of the app. 

Cheers, 
Colin

**Reply by hedjour** [reply]
Many thanks for the twoo possibilities
